### DISCUSSION TITLE
GO Using a Different MIDI Device Name in Later Versions?

### DISCUSSION THREAD

**Discussion by OrgelSpieler** [GrandOrgue/grandorgue#872] [Help] [answered]
I've had been successfully running GO verions 0.3.1.2330-156.3 for sometime on a Win 7 Platform with an RME AIO sound card. I decided to get my instrument on the latest version 3.4.2-1.  After installing the upgrade, GO no longer responded to my midi instrument. I came to find out that the upgrade uses a different device name for my sound card than my older GO version.   Under my old version (0.3.1.2330-156), my midi input device was shown as AIO Midi 0.  When I did the upgrade (3.4.2-1), my midi device shows as winmm:AIO Midi 0 (note the prefix "winmm" has been added).  Since all of my midi settings were set up for AIO Midi 0, the system could not map the inputs because it was expecting winmm:AIO Midi 0.

I never implemented any of the intermediate versions, so I don't if this change might have occurred on an earlier version other than the latest release.  I looked through the release notes and was unable to find any mention of this. 

If all I had to do was redo the assignments for the manuals and pedal, it would be no big deal. But I have assignments for 156 physical stops, couplers and pistons (both send and receive). To redo all of that is quite an undertaking. 

Is there anyway to force the system to use the older midi names?
If the new names must be used, is there a file that can be edited so the change can be done efficiently?

**Comment by oleg68** [comment]
> Is there anyway to force the system to use the older midi names?

No it is not possible.

But there are some plans of enchancing matching of MIDI device names in #703.

> If the new names must be used, is there a file that can be edited so the change can be done efficiently?

GrandOrgueConfig and GrandOrgueData/*.cmb

But direct changing of these files is not supported

**Reply by OrgelSpieler** [reply]
Thank you for such a quick response.  

Am I correct to conclude that since GrandOrgueData and GrandOrgueConfig don't exist in an editable format,  I  have no recourse but to re-enter all of my data?  If so, that is an unpleasant task.  What changed in the release that GO retrieves a different name for the MIDI device especially since the midi is retrieved over the bus and not a USB port?

Have others had to re-enter their config data upon an upgrade or is this unique to my configuration?

**Comment by oleg68** [comment, marked as answer]
> Am I correct to conclude that since GrandOrgueData and GrandOrgueConfig don't exist in an editable format, I have no recourse but to re-enter all of my data? If so, that is an unpleasant task. 

You can make these file editable, but please save a backup copy of them before any manipulations. If something goes wrong you will be able to restore them.

These files are gzip archives. You can unpack the text config files in ini-format from the archives and edit them. But we do not provide any documentation about them.

> What changed in the release that GO retrieves a different name for the MIDI device especially since the midi is retrieved over the bus and not a USB port?

Most probably 

- 0.3.1.2341-9.os (2021-08-07)
- Changed the sound device name format to ``Subsys: Api: Device`` https://github.com/oleg68/GrandOrgue/issues/48

It was intended to change device names only for audio, not for midi. But changing the midi names occured as an undesired side effect.

> Have others had to re-enter their config data upon an upgrade or is this unique to my configuration?

Nobody others complained. Maybe your uniqueness is having as many switches to reconfig? Reassigning only a few events for the pedal and the manuals is not so hard.

**Reply by OrgelSpieler** [reply]
Thank you.

**Comment by OrgelSpieler** [comment]
Thank you for the detailed response and explanation.  I understand that sometimes unintended consequences just happen.
I think I can handle the file edit - I've done similar things in a past life.   Since compression utilities allow for a lot of configuration options is there any kind of special format the GO is expecting when I compress the file?

**Comment by oleg68** [comment]
> is there any kind of special format the GO is expecting when I compress the file?

Yes, `gzip`.

But in most cases you may put uncumpressed text files with the same names.

**Reply by OrgelSpieler** [reply]
Thanks for the info on the files zip algorithm.  I successfully completed the edit of the files and was even able to clean up old stuff that had been lurking around from previous hardware configurations.  Once I understood how the files were structured, it was rather easy and quick to do the edit.  You saved me a lot of time.  

The only thing I encountered that I cannot explain was related to use of Remote Desktop.  If I tried to edit the files and start GO through Windows Remote Desktop, I encountered problems with GO detecting my midi devices (it would think none were there).  If I did the edits directly on the organ's computer, I had no problems.  That had me stymied a bit, but once I figured out that was a problem, it only took a few minutes to restore an old version, upgrade and then do the files edits.

Your help saved me a lot of time.

**Comment by jason-barba** [comment]
> > What changed in the release that GO retrieves a different name for the MIDI device especially since the midi is retrieved over the bus and not a USB port?
> 
> Most probably
> 
> * 0.3.1.2341-9.os (2021-08-07)
> * Changed the sound device name format to `Subsys: Api: Device` [Make lazy audio system initialisation oleg68/GrandOrgue#48](https://github.com/oleg68/GrandOrgue/issues/48)
> 
> It was intended to change device names only for audio, not for midi. But changing the midi names occured as an undesired side effect.

Does the midi name change also affect the Debian/Ubuntu build of latest version?

**Reply by larspalo** [reply]
Yes it does.

**Reply by jason-barba** [reply]
OK. Good to know.
